### PR TITLE
fix: fix Sphinx syntax and build docs in CI

### PR DESCRIPTION
**Pull request by danking**
We tested the docs but did not verify the rST syntax was correct. Now we both test and verify the syntax is correct and all links are valid.

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 # You can set these variables from the command line, and also
 # from the environment for the first two.
-SPHINXOPTS    ?= -W --keep-going
+SPHINXOPTS    ?= --fail-on-warning --keep-going
```
Comment (danking) [inline docs/Makefile:6]: `-W` is short for this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -51,6 +51,14 @@ jobs:
...
         working-directory: docs/
 
+      - name: Ensure docs build - PyVortex
+        run: |
+          set -ex
+
+          (cd pyvortex && uv run maturin develop)
+
+          cd docs
+          uv run sphinx-build -M html . _build --fail-on-warning --keep-going
```
Comment (AdamGS) [inline .github/workflows/ci.yml:61]: can we just use the makefile here?

**Review by AdamGS — APPROVED** [review]

**Review by lwwmanning — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -51,6 +51,14 @@ jobs:
           make clean && make doctest
         working-directory: docs/
 
+      - name: Ensure docs build - PyVortex
+        run: |
+          set -ex
+
+          (cd pyvortex && uv run maturin develop)
```
Comment (robert3005) [inline .github/workflows/ci.yml:58]: we should have already built it?

Comment (danking) [reply .github/workflows/ci.yml:58]: I think so? `uv run pytest` should have the side effect of building and installing it, which is what develop does.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (danking) [reply .github/workflows/ci.yml:61]: Sure, I can do that in a follow up PR since this already slipped in.

Comment (danking) [reply .github/workflows/ci.yml:61]: https://github.com/spiraldb/vortex/pull/1156

**Pull request merged**